Chore/upgrade turbo

cbliard opened the pull request:
# Ticket

None

# What are you trying to accomplish?

I tried using turbo morph instead of a custom rendering with iodiomorph to update the progress popover. It did not work.

The turbo has been upgraded and a stimulus controller has been merged in the process, which is still valuable, hence this pull request.

# What approach did you choose and why?

The turbo lib have been upgraded to 8.0.10 because there was a bug which prevented morph from working when triggered from javascript. It was fixed in this version.

Also using morph needed some `data-turbo-permanent` on the inputs getting focus so that the value and the focus are kept, which is why the 2 controllers have been merged.

# Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation in Lookbook (patterns, previews, etc)
- [x] Tested major browsers (Chrome, Firefox, Edge, ...)

cbliard committed:
Handle focus in one unique controller

Also the error `Missing target element "fieldToFocus" for
"work-packages--progress--focus-field" controller` sometimes happens
when there is no field to focus, and stimulus does not like
unspecified targets. Switching to a simpler data attribute as well to
avoid that error.

cbliard committed:
Upgrade turbo to 8.0.10

After having tried to use morphing in the progress popover, I found that
it's not working out quite well. The comment has been updated to explain
why.

cbliard committed:
Remove dead code

aaron-contreras reviewed (approved):
Nicely done inlining the focus controller too

cbliard committed:
Remove no longer needed Cuprite patch

https://github.com/rubycdp/cuprite/pull/217 has been merged, `save_path`
is correctly taken into account now.

cbliard committed:
Fix flakiness in spec work_package_activity_spec.rb

Two different changes were made: one changing work, one changing
remaining work. In activities, the entry was created on the first
change, and then updated with new content on the second change.

When `within("activity-entry", text: admin.name)` was used, the first
entry could be returned depending on how fast the server responds. That
entry would then become stale when the entry is updated with the new
content, leading to the test failing with the following error: expected
to find visible list item with text "% Complete set to 95%" within
"<<ERROR>>", "<<ERROR>>", "<<ERROR>>", "<<ERROR>>", "<<ERROR>>",
"<<ERROR>>", which matched the selector but not all filters.

"<<ERROR>>" is what is returned when the element cannot return its text
content.

This happened in https://github.com/opf/openproject/actions/runs/10926176246/job/30329450428

This flakiness error was fixed by doing only one modification to the
work package, so that the entry is created once and never updated.

cbliard commented:
The one failing test comes from `dev` branch and has been fixed in #16757
Merging
